### ISSUE TITLE
Support wasm-c-api

### ISSUE THREAD

**Issue by syrusakbary** [titzer/wizard-engine#29]
It would be great if the Wizard Engine exported the wasm-c-api.

We recently opened Wasmer to allow usage of more runtimes under the hood, and this would allow users using Wizard Engine as a backend for Wasmer.
We'd be happy to sponsor this effort if it makes prioritizing it easier :)

**Comment by titzer** [comment]
Can you share a little about your use case? Is it to link the engine into a program that implements your platform functionality? Or is it to link it into arbitrary programs?

**Comment by syrusakbary** [comment]
Of course. We want to start trialing more runtimes and allow others to switch runtimes easily. Wasm-C-API serves both of this use case.

We can benchmark how other runtimes perform and also we can allow any program to use the engine of their preference

**Comment by titzer** [comment]
Implementing the C API is a lot of work and would be blocked on significant compiler work in Virgil, so I won't be able to prioritize this. Is there another way you can test runtimes short of embedding them?

**Comment by syrusakbary** [comment]
> Is there another way you can test runtimes short of embedding them?

I think embedding via the C-API is probably the easiest of all, At the end we need to integrate through the syscalls (eg: WASIX) and that requires runtime integration
